core/src/SYCL/Kokkos_SYCL_Instance.cpp

```diff
@@ -187,6 +192,60 @@ void SYCLInternal::finalize() {
   m_queue.reset();
 }
 
+void* SYCLInternal::scratch_space(
+    const Kokkos::Experimental::SYCL::size_type size) {
+  const size_type sizeScratchGrain =
+      sizeof(Kokkos::Experimental::SYCL::size_type);
+  if (verify_is_initialized("scratch_space") &&
+      m_scratchSpaceCount * sizeScratchGrain < size) {
+    m_scratchSpaceCount = (size + sizeScratchGrain - 1) / sizeScratchGrain;
+
+    using Record = Kokkos::Impl::SharedAllocationRecord<
+        Kokkos::Experimental::SYCLDeviceUSMSpace, void>;
+
+    if (nullptr != m_scratchSpace)
+      Record::decrement(Record::get_record(m_scratchSpace));
+
+    Record* const r = Record::allocate(
+        Kokkos::Experimental::SYCLDeviceUSMSpace(*m_queue),
+        "InternalScratchSpace", (sizeScratchGrain * m_scratchSpaceCount));
```

Comment:
Make sure you prefix that label
```suggestion
        "Kokkos::InternalScratchSpace", (sizeScratchGrain * m_scratchSpaceCount));
```

Reply:
Just trying to be consistent with the other backends. :wink: